lib/gems/pending/util/mount/miq_s3_session.rb

```diff
@@ -0,0 +1,63 @@
+require 'util/mount/miq_generic_mount_session'
+require 'aws-sdk'
+
+class MiqS3Session < MiqGenericMountSession
+  def initialize(log_settings)
+    log_header = "MIQ(#{self.class.name}-initialize)"
+    logger.debug("#{log_header} initialize: log_settings are #{log_settings}")
+    super(log_settings)
+    logger.debug("#{log_header} initialize: @settings are #{@settings}")
+    raise "username is a required value!" if @settings[:username].nil?
+    raise "password is a required value!" if @settings[:password].nil?
+    raise "region is a required value!" if @settings[:region].nil?
+    @s3 = s3
+  end
+
+  def s3
+    @s3 ||= Aws::S3::Resource.new(:region => @settings[:region], :access_key_id => @settings[:username], :secret_access_key => @settings[:password])
+  end
+
+  def connect
+    _scheme, _userinfo, @host, _port, _registry, @mount_root, _opaque, _query, _fragment = URI.split(URI.encode(@settings[:uri]))
+    @mount_path = @mount_root.split("/")[0..1].join("/")
+    super
+  end
+
+  def disconnect
+    return if @mnt_point.nil?
+    FileUtils.rm_rf(@mnt_point) if File.exist?(@mnt_point)
+  end
+
+  def mount_root
+    File.join(@mnt_point, (@mount_root.split("/") - @mount_path.split("/")))
```

Comment:
When dealing with file paths, wouldn't it be better to use path-specific operations like `filename`, `basename`, or even the `Pathname` or `URI` classes - as opposed to splitting on "/"? The same applies to the other path manipulations in this file.

Reply:
Yes.  I couldn't come up with anything that did what I wanted so far so I did it from scratch.  I can go back and look again.